org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/serializer/SerializerFragment2.xtend

```diff
@@ -456,12 +460,29 @@ class SerializerFragment2 extends AbstractStubGeneratingFragment {
 				
 				@�Inject�
 				protected void init(�IGrammarAccess� access) {
-					grammarAccess = (�grammar.grammarAccess�) access;
-					�FOR group : allAmbiguousTransitionsBySyntax�
-						match_�group.identifier� = �group.elementAlias.elementAliasToConstructor�;
-					�ENDFOR�
+					�IF partitions.size > 1�
+						�FOR partition : partitions.indexed�
+							init�partition.key�(access);
+						�ENDFOR�
+					�ELSE�
+						grammarAccess = (�grammar.grammarAccess�) access;
+						�FOR group : allAmbiguousTransitionsBySyntax�
+							match_�group.identifier� = �group.elementAlias.elementAliasToConstructor�;
+						�ENDFOR�
+					�ENDIF�
 				}
 				
+				�IF partitions.size > 1�
+					�FOR partition : partitions.indexed�
+						private void init�partition.key�(�IGrammarAccess� access) {
+							grammarAccess = (�grammar.grammarAccess�) access;
```

Comment:
Hmm I'm surprised.
I thought we'd set the grammar access only once in the main-init method. The init0 etc shouldn't take any arguments.